frameworks/helloworld/tests/test_networking.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+
+import sdk_install as install
+import sdk_plan as plan
+import sdk_spin as spin
+
+from tests.config import (
+    PACKAGE_NAME
+)
+
+
+def setup_module(module):
+    install.uninstall(PACKAGE_NAME)
+    options = {
+        "service": {
+            "spec_file": "examples/cni.yml"
+        }
+    }
+
+    install.install(PACKAGE_NAME, 1, additional_options=options)
+    print('done installing...')
+
+
+@pytest.mark.sanity
+def test_deploy():
+    deployment_plan = plan.get_deployment_plan(PACKAGE_NAME).json()
+    print("deployment_plan: " + str(deployment_plan))
+
+    assert(len(deployment_plan['phases']) == 1)
+    assert(deployment_plan['phases'][0]['name'] == 'all-deploy')
+    assert(len(deployment_plan['phases'][0]['steps']) == 2)
+
+
+@pytest.mark.sanity
+def test_joins_overlay_network():
```

Comment:
It does. Unfortunately the actual logic is in the YAML definition itself (see `cni.yml` in this PR), since there's no real sensible way to get information about whether a property holds for a task's environment (such as "is my IP address what I expect?") other than constructing the command so that it fails if that property is false. In this case, the command greps for `9\.[0-9\.]+` in the output of `ip addr show`. If the plan makes it to `COMPLETE`, the task finished and all is well.